src/python/tests/test_configuration_manager.py

```diff
@@ -0,0 +1,653 @@
+                                  'default': 'False'}})
+        result = await get_category_item_value_entry(category_name='boolean', item_name='info')
+        assert result == 'False'
+
+        await set_category_item_value_entry(category_name='boolean',
+                                            item_name='info', new_value_entry='True')
+        result = await get_category_item_value_entry(category_name='boolean', item_name='info')
+        assert result == 'True'
+
+    async def test_get_category_item_value_entry_dne(self):
+        """
+        Test that None gets returned when either category_name and/or item_name don't exist
+        :assert:
+            1. Assert None is returned when item_name does not exist
+            2. Assert None is returned when category_name does not exist
+        """
+        await create_category(category_name='boolean', category_description='boolean type',
+                              category_value={
+                                  'info': {
+                                      'description': 'boolean type with default False',
+                                      'type': 'boolean',
+                                      'default': 'False'}
+                              })
+        result = await get_category_item_value_entry(category_name='boolean', item_name='data')
+        assert result is None
+
+        result = await get_category_item_value_entry(category_name='integer', item_name='info')
+        assert result is None
+
+    async def test_get_category_item(self):
+        """
+        Test that get_category_item returns all the data in configuration.
+        value for a specific category_name
+        :assert:
+            1. Information in configuration.value match the category_values declared
+            2. When updating value, the data retrieved for default does not change,
+            while for value it it does
+        """
+        await create_category(category_name='boolean', category_description='boolean type',
+                              category_value={
+                                  'info': {
+                                      'description': 'boolean type with default False',
+                                      'type': 'boolean',
+                                      'default': 'False'}
+                              })
+        result = await get_category_item(category_name='boolean', item_name='info')
+        assert result['description'] == 'boolean type with default False'
+        assert result['type'] == 'boolean'
+        assert result['default'] == 'False'
+        assert result['value'] == 'False'
+
+        await set_category_item_value_entry(category_name='boolean',
+                                            item_name='info', new_value_entry='True')
+        result = await get_category_item(category_name='boolean', item_name='info')
+        assert result['default'] == 'False'
+        assert result['value'] == 'True'
```

Comment:
can not this be verified in `async def test_get_category_item_value_entry(self):` itself?

Reply:
`test_get_category_item_value_entry` (renamed to `test_get_set_category_item_value_entry`) tests  the `get_category_item_value_entry` which returns the value in `statistics.value` json object. While `test_get_category_item` tests `get_category_item`, which the json object for a given `item_name`.